### ISSUE TITLE
[clang-format-16] IntegerLiteralSeparator for binary literals broken

### ISSUE THREAD

**Issue by h-2** [llvm/llvm-project#60948]
With the following options:
```yml
IntegerLiteralSeparator:
  Binary: 4
  Decimal: 0
  Hex: 0
```

Clang-Format complains:
```console
error: code should be clang-formatted [-Wclang-format-violations]
    constexpr uint8_t to_op_rank() const noexcept { return rank & 0b1111; }
                                                                    ^
```

This is a false positive, because `0b'1111` is not a valid bianry literal. Interestingly, `clang-format -i` also does not "fix" this.

**Comment by owenca** [comment]
> Interestingly, `clang-format -i` also does not "fix" this.

Can you attach the input file and the exact command you were using? As you said, running clang-format by itself doesn't reproduce the problem you saw.

**Comment by JVApen** [comment]
@owenca  We are also seeing this problem.
I've logged https://github.com/llvm/llvm-project/issues/61048 for a similar issue, possibly it is a duplicate.
It contains a small reproduction scenario.

**Comment by rymiel** [comment]
This was fixed by b4243bb611a30137f3f1a7f0e531f91d331ab4be (b98d95984f0eb81a5f06eeaa3df29bbb1834a04b on release/16.x)
